### PR TITLE
Clade mismatch

### PR DESCRIPTION
**Pull request by nick-pestell**
This PR implements a bug fix related to a data mismatch, where the clade in cattle data in the MDWH is occasionally incorrectly assigned. 

I have implement a new function in `consistify.py` called `clade_correction()`. This basically reassigns the clade in the cattle data frame to match the clade in the wgs data. 

I have modularised the consistify logic so that cattle, movement and wgs data are "consitified" in one function, the clade is corrected in another function and finally, nans are removed from movement csv in a third function, `fix_movements()`. This are then all called within `process_datasets()`.  This is an improvement on the previous logic in which `fix_movements()` and `consistify()` were part of the same function.

I have also written additional unit tests.

### CODE REVIEW

File: btb_phylo.py [hunk 1]
```diff
@@ -154,7 +154,7 @@ def consistify_samples(results_path, cat_mov_path):
                                                        consistified_movement_filepath, metadata_path)
     # copy cattle and movement csvs to metadata
     shutil.copy(cattle_filepath, os.path.join(metadata_path, "cattle.csv"))
-    shutil.copy(cattle_filepath, os.path.join(metadata_path, "movement.csv"))
+    shutil.copy(movements_filepath, os.path.join(metadata_path, "movement.csv"))
```
Comment (CameroneN) [inline btb_phylo.py:157]: Very minor thing but could make it consistent with / without the s?

Comment (nick-pestell) [reply btb_phylo.py]: done

File: btbphylo/consistify.py [hunk 2]
```diff
@@ -55,22 +54,56 @@ def consistify(wgs, cattle, movements):
...
+    movement_consist = movement[movement.SampleName.isin(consist_samples)].copy()
+    return wgs_consist, cattle_consist, movement_consist,\
+        missing_wgs, missing_cattle, missing_movement
+
+def clade_correction(wgs, cattle):
+    """
+        Ensures that the clade assigment in cattle csv matches the clade in
+        WGS data. Assumes wgs clade is correct and overwrites the cattle calde
+        if there is a mismatch. This feature corrects an error where the wrong 
+        clade is assigned in the MDWH.
```
Comment (ellisrichardj) [inline btbphylo/consistify.py:66]: Slightly worrying that there's a mismatch, but you're right in assuming the WGS version will be correct.  From a quick assessment of the list that Cam sent, it seems to be linked to the lower quality samples, so they would be excluded from Viewbovine anyway.  Essentially, they don't have a clade in MDWH, but WGS has assigned one, despite the data being low quality

Comment (nick-pestell) [reply btbphylo/consistify.py:66]: Ok, from my memory we were experiencing some issues within the app) as a result of the clade mismatch (I think in some cases genetic related samples were coming through but with the wrong clade, so perhaps a few of the offending samples slip through above the filter thresholds. Which parameter were you looking at when deciding they are low quality?

I feel that either way it's not going to hurt correcting for this and is probably the safest /easiest thing to do , at least in the short term. Until we get to the bottom of the source of the mismatch ...

**Review by ellisrichardj — APPROVED** [review]
Great to get the fix done

**Pull request merged**